Sources/_StringProcessing/RegexDSL/DSL.swift

```diff
@@ -11,6 +11,19 @@
 
 import _MatchingEngine
 
+// A convenience protocol for builtin regex components that are initialized with
+// a `DSLTree` node.
+internal protocol _BuiltinRegexComponent: RegexComponent {
+  var regex: Regex<Match> { get }
+  init(_ regex: Regex<Match>)
```

Comment:
Do we want this defaulted with `self.regex = regex`? Or even just have this in an extension?

Reply:
Protocol initializers can only do delegate initialization, so I don’t think we can.  There’s no way to express a formal guarantee that a protocol property will be a stored property.